lib/rack/set_x_forwarded_proto_header.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Rack
+
+  # Middleware to set the X-Forwarded-Proto header to the value
+  # of another header.
+  #
+  # This header can be used to ensure the scheme matches when comparing
+  # request.origin and request.base_url for CSRF checking, but Rack
+  # expects that value to be in the X_FORWARDED_PROTO header.
+  #
+  # For example, AWS Cloudfront sets a CloudFront-Forwarded-Proto header
+  # https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/adding-cloudfront-headers.html#cloudfront-headers-other
```

Comment:
I would prefer if the documentation and specs removed all examples with CloudFront.  Please switch to a generic key so that we are not giving a preference to a specific vendor.